src/emitters/astroobject.cpp

```diff
@@ -0,0 +1,256 @@
+                      "can be specified at the same time!'");
+
+            ScalarVector3f direction(dr::normalize(props.get<ScalarVector3f>("direction")));
+            m_direction = direction;
+            auto [up, unused] = coordinate_system(direction);
+
+            m_to_world = ScalarTransform4f::look_at(0.0f, ScalarPoint3f(direction), up);
+            dr::make_opaque(m_to_world);
+        }
+
+        Float angular_diameter = SUN_ANGULAR_DIAMETER;
+        if (props.has_property("angular_diameter")) {
+            angular_diameter = props.get<Float>("angular_diameter");
+        }
+        Float m_angular_radius = dr::deg_to_rad(angular_diameter / 2.f);
+        m_angular_radius_cos = dr::cos(m_angular_radius);
+        if (dr::any((1.0 <= m_angular_radius_cos) | (m_angular_radius_cos < 0.f))) {
+            Throw("Invalid angular diameter specified! (must be in ]0, 180[°)");
+        }
+
+        m_omega = 2.f * dr::Pi<Float> * (1 - m_angular_radius_cos);
+        m_irradiance = props.texture_d65<Texture>("irradiance", 1.f);
+        Log(
+            Debug,
+            "angular_radius: %s; angular_radius_cos: %s; solid angle omega: %s",
+            m_angular_radius, m_angular_radius_cos, m_omega
+        );
+
+        if (m_irradiance->is_spatially_varying())
+            Throw("Expected a non-spatially varying irradiance spectra!");
+
+        m_needs_sample_3 = false;
+
+        m_flags      = +EmitterFlags::Infinite;
+        dr::set_attr(this, "flags", m_flags);
+    }
+
+    void traverse(TraversalCallback *callback) override {
+        callback->put_object("irradiance", m_irradiance.get(), +ParamFlags::Differentiable);
+        callback->put_parameter("to_world", *m_to_world.ptr(), +ParamFlags::NonDifferentiable);
+    }
+
+    void set_scene(const Scene *scene) override {
+        if (scene->bbox().valid()) {
+            m_bsphere = scene->bbox().bounding_sphere();
+            m_bsphere.radius =
+                dr::maximum(math::RayEpsilon<Float>,
+                        m_bsphere.radius * (1.f + math::RayEpsilon<Float>) );
+        } else {
+            m_bsphere.center = 0.f;
+            m_bsphere.radius = math::RayEpsilon<Float>;
+        }
+    }
+
+    Spectrum eval(const SurfaceInteraction3f & si,
+                  Mask active) const override {
+        MI_MASKED_FUNCTION(ProfilerPhase::EndpointEvaluate, active);
+
+        Float cos_dirwwi = dr::dot(si.wi, m_direction);
+
+        return depolarizer<Spectrum>(m_irradiance->eval(si, active) / m_omega) &
+            (cos_dirwwi > m_angular_radius_cos);
+    }
+
+    std::pair<DirectionSample3f, Spectrum>
+    sample_direction(const Interaction3f &it, const Point2f & sample,
+                     Mask active) const override {
+        MI_MASKED_FUNCTION(ProfilerPhase::EndpointSampleDirection, active);
+
+        Vector3f local_dir = warp::square_to_uniform_cone(sample, m_angular_radius_cos);
```

Comment:
For consistency with the rest of variable naming, I'd prefer if you would name this one `m_cos_angular_radius`.